docs/controller_plugins.md

```diff
@@ -106,6 +106,20 @@ Finally, run any workflow. You'll see "hello" printed in the plugins sidecar's l
 kubectl -n argo logs deployment/workflow-controller -c hello-controller-plugin
 ```
 
+## Security
+
+Besides the [general plugin security considerations](plugins.md#security), controller plugins require careful thought to
+implement securely.
+
+Keep in mind:
+* Since it resides on the same pod as the workflow controller, it shares resources with the workflow controller (cpu, 
```

Comment:
As currently implemented, I'm not sure that a controller plugin _has_ to reside on the workflow-controller pod. Seems like you could create a deployment and a service and just configure the plugin to point at that service.

Reply:
We could probably force developers to put their plugins on the workflow-controller pod by hard-coding the domain name to be `localhost`. I think that would be more secure in a way because only the controller admin could add plugins.

But that would prevent developers from having scalable plugins on their own pod(s) (which would also protect the controller's resources). The downside of non-sidecar controller plugins is that (I think) non-admins could install them.